libsolidity/interface/CompilerStack.cpp

```diff
@@ -1392,6 +1437,7 @@ string CompilerStack::createMetadata(Contract const& _contract) const
 	static vector<string> hashes{"ipfs", "bzzr1", "none"};
 	meta["settings"]["metadata"]["bytecodeHash"] = hashes.at(unsigned(m_metadataHash));
 
+	meta["settings"]["viaIR"] = m_viaIR;
```

Comment:
Maybe only set it if it is true?

Reply:
But the weird thing is, since it is always set currently (even if false), it should trigger some test failures. Is it possible that `false` keys are removed?

Reply:
since the compiler version is part of the metadata, it is not tested in out-update tests, I think, only in the c++-based metadata test.